aiida/restapi/translator/base.py

```diff
@@ -476,15 +422,19 @@ def get_formatted_result(self, label):
         if self._total_count > 0:
             for res in self.qbobj.dict():
                 tmp = res[label]
-                if self._default_user_projections:
-                    tmp['user_email'] = res['user']['email']
+
+                # Note: In code cleanup and design change, remove this node dependant part
+                # from base class and move it to node translator.
+                if self._result_type in ['with_outgoing', 'with_incoming']:
+                    tmp['link_type'] = res[self.__label__ + '--' + label]['type']
```

Comment:
Question: Does the logic of separating these two parts by `--` come from the querybuilder?

Reply:
Yes. Its the syntax of querybuilder to get edge_properties

Reply:
ok